R/sass.R

```diff
@@ -55,63 +55,60 @@ sass <- function(input = NULL, options = sass_options(), output = NULL,
     stop("sass(write_attachments=TRUE) cannot be used when output=NULL")
   }
 
-  use_cache <- isTRUE(cache_options[["cache"]])
-  cache_file <- if (use_cache) {
-    cache_file_path(
-      normalizePath(cache_options[["cache_dir"]], mustWork = TRUE),
-      input,
-      options
-    )
-  }
-
   css <- NULL
   layer <- NULL
-  if (use_cache && file.exists(cache_file)) {
-    # No need to read cache file if output is specified
-    if (!is.null(output)) {
-      fs::file_copy(cache_file, output, overwrite = TRUE)
-      if (isTRUE(write_attachments == FALSE)) {
+
+  if (!is.null(cache)) {
+    cache_key <- sass_hash(list(
+      input,
+      # Detect if any attachments have changed - note that if the attachment is
+      # a directory, it will detect if files are added or removed, but it will
+      # not detect if a file in the directory is modified.
+      if (is.list(input) && !is.null(input$file_attachments)) file.mtime(input$file_attachments),
```

Comment:
If we decided to include `file_attachments` in the hash here, we'll need to `layer <- extract_layer(input)` first, then `file.mtime(layer$file_attachments)`